### PR TITLE
Enable scrolling and page-in previous texts as necessary when scrolling

### PR DESCRIPTION
**Pull request by willhite**
Also:
Move debug function into utils package so I can use it from other code when testing.
Add /q command to enable quitting without cntl-c.

### CODE REVIEW

File: go/util/dbg/debug.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+func NewLogger(fp string) *log.Logger {
+    f, err := os.OpenFile(fp, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+    d.PanicIfError(err)
+    pid := strconv.FormatInt(int64(os.Getpid()), 10)
+    return log.New(f, pid + ": ", 0644)
+}
+
+func GetLogger() *log.Logger {
```
Comment (aboodman) [inline go/util/dbg/debug.go]: Why would you ever get or set a logger?

Comment (aboodman) [inline go/util/dbg/debug.go]: This doesn't seem to be used, recommend removing.

File: go/util/dbg/debug.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+    "fmt"
+    "log"
+    "os"
+    "github.com/attic-labs/noms/go/d"
+    "strconv"
+)
+
+var (
```
Comment (aboodman) [inline go/util/dbg/debug.go]: Hm. I usually would support pulling something like this out into a utility but in this case, it has a pretty big side-effect. Any code which links against this will start writing to that temp file.

I think I prefer to let clients create a logger as they see fit and not have this utility.

File: samples/go/ipfs-chat/main.go [hunk 3]
```diff
@@ -34,63 +42,119 @@ var (
...
 	}
 
-	active = nextIndex
-	return nil
+	var msg string
+	for m := range dp.msgChan {
+		if m == emptyMessagesResult {
+			return countLines(buf)
+		}
+		if dp.shouldFilter {
```
Comment (aboodman) [inline samples/go/ipfs-chat/main.go]: There is a fancier/more efficient way to do this that does real set intersection in Noms. We should use it, but can be a different PR.

Comment (aboodman) [inline samples/go/ipfs-chat/main.go]: (the downside of what you're doing here is that if the the thing you're looking for is a needle in a haystack we'll load all the data into memory to search for it ... whereas using noms we could seek right to it)

Comment (willhite) [inline samples/go/ipfs-chat/main.go]: This is the way it was done originally and I didn't want to take the time now to figure out how to do it more efficiently. I agree this needs to be more efficient.

File: go/util/dbg/debug.go (hunk 2 above)
Comment (willhite) [inline go/util/dbg/debug.go]: This is kind of a weird situation because it was really useful to throw a dbg.Debug() line in a couple of places in the noms code while developing but I wouldn't expect any code that actually ships to have calls to this function in it. I guess I could put it into a utils package underneath the ipfs-chat directory for now. I could also add an Enabled flag that would cause it to nothing unless some program explicitly enabled it.

File: go/util/dbg/debug.go (hunk 1 above)
Comment (willhite) [inline go/util/dbg/debug.go]: You need to get a logger in order to use it. I thought SetLogger() would be useful if someone wanted to log to a different file. The nice thing about this is any function can grab the logger, it doesn't have to be passed around.
